packages/frontend/component/src/ui/date-picker/week-date-picker.tsx

```diff
@@ -0,0 +1,248 @@
+import { ArrowLeftSmallIcon, ArrowRightSmallIcon } from '@blocksuite/icons';
+import clsx from 'clsx';
+import dayjs from 'dayjs';
+import {
+  type ForwardedRef,
+  type HTMLAttributes,
+  memo,
+  useCallback,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+
+import { IconButton } from '../button';
+import * as styles from './week-date-picker.css';
+
+export interface WeekDatePickerHandle {
+  /** control cursor manually */
+  setCursor?: (cursor: dayjs.Dayjs) => void;
+}
+
+export interface WeekDatePickerProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, 'onChange'> {
+  value?: string;
+  onChange?: (value: string) => void;
+  handleRef?: ForwardedRef<WeekDatePickerHandle>;
+}
+
+// TODO: i18n
+const weekDays = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+// const weekDays = ['日', '一', '二', '三', '四', '五', '六'];
+const format = 'YYYY-MM-DD';
+
+export const WeekDatePicker = memo(function WeekDatePicker({
+  value,
+  onChange,
+  className,
+  handleRef,
+  ...attrs
+}: WeekDatePickerProps) {
+  const weekRef = useRef<HTMLDivElement | null>(null);
+
+  const [cursor, setCursor] = useState(dayjs(value));
+  const [range, setRange] = useState([0, 7]);
+  const [dense, setDense] = useState(false);
+  const [allDays, setAllDays] = useState<dayjs.Dayjs[]>([]);
+  const [viewPortSize, setViewPortSize] = useState(7);
+
+  useImperativeHandle(handleRef, () => ({
+    setCursor,
+  }));
+
+  const displayDays = allDays.slice(...range);
+
+  const updateRange = useCallback(
+    (newRange: [number, number]) => {
+      if (range && newRange[0] === range[0] && newRange[1] === range[1]) return;
+      setRange(newRange);
+    },
+    [range]
+  );
+
+  const onNext = useCallback(() => {
+    if (viewPortSize === 7) {
+      setCursor(cursor.add(1, 'week'));
+    } else if (range[1] === 7) {
+      setCursor(cursor.add(1, 'week'));
+      updateRange([0, viewPortSize]);
+    } else {
+      const end = Math.min(range[1] + viewPortSize, 7);
+      const start = Math.min(range[0] + viewPortSize, end - viewPortSize);
+      updateRange([start, end]);
+    }
+  }, [cursor, range, updateRange, viewPortSize]);
+  const onPrev = useCallback(() => {
+    if (viewPortSize === 7) {
+      setCursor(cursor.add(-1, 'week'));
```

Comment:
When I use the keyboard to go to the previous or next week, the focus will be lost and can't action continuously

Reply:
Yes, the implementation of keyboard navigation requires a better solution, so I marked it as `// TODO`.